### PR TITLE
Add the AutoFDO tutorial.

### PR DESCRIPTION
**Pull request by rcaballeromx**
The AutoFDO tutorial was hidden in the AutoFDO feature content.
Edited the content and formatted it in RST.
Integrated the content as an independent tutorial and added a link to it in the
feature content on the new website.

Signed-off-by: Rodrigo Caballero <rodrigo.caballero.abraham@intel.com>

### CODE REVIEW

File: source/clear-linux/tutorials/autofdo.rst [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+* fpredictive-commoning
+* fgcse-after-reload
+* ftree-loop-vectorize
+* ftree-loop-distribute-patterns
+* ftree-slp-vectorize
+* fvect-cost-model
+* ftree-partial-pre
+* fipa-cp-clone
+
+We applying this flag to our example code and run it:
```
Comment (rcaballeromx) [inline source/clear-linux/tutorials/autofdo.rst]: +1

File: source/clear-linux/tutorials/autofdo.rst [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+We see the execution reduced by 59.6 percent. An impressive performance
+increase for a single optimization flag based on only the static analysis of
+the code. However, the execution time provides no input that can tell us how
+the code is behaving for the user.
+
+The FDO technology allows us to gain feedback on which parts of the code are
+never executed or which parts are more worthwhile to optimize.
+
+.. _FDO:
```
Comment (rcaballeromx) [inline source/clear-linux/tutorials/autofdo.rst]: We must keep consistency. All labels should be all in lowercase.

File: source/clear-linux/tutorials/autofdo.rst [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+   to the additional instrumentation code executed.
+
+#. Build an optimized version of the program by using the collected execution
+   profile to guide the optimizations, the FDO build.
+
+The instrumentation and FDO builds are tightly coupled. GCC requires both
+builds use the same in-line decisions and similar optimization flags to
+ensure that the :abbr:`CFG (Control-Flow Graph)` instrumented in the
+instrumentation build matches the CFG annotated with the profile data in the
+FDO build.
```
Comment (rcaballeromx) [inline source/clear-linux/tutorials/autofdo.rst]: No abbr is needed if the abbreviation was written out previously on the document or if abbr was used once before.

File: source/clear-linux/tutorials/autofdo.rst [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+Enter AutoFDO
+=============
+
+To overcome the limitations of the current FDO model, we proposed the use of
+AutoFDO. The AutoFDO tool uses `perf`_ to collect sample profiles. A
+standalone tool converts the :file:`perf.data` file into the `gcov` format.
+See the `tool's source code`_ for details.
+
+This new model skips the instrumentation step. Instead, it uses a sampling-
```
Comment (rcaballeromx) [inline source/clear-linux/tutorials/autofdo.rst]: I agree, not having the constant comparison allows us to concentrate on explaining the actual feature.

File: source/clear-linux/tutorials/autofdo.rst [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+-------------------------
+
+AutoFDO needs a :file:`perf.data` file to capture the `BR_INST_RETIRED:TAKEN`
+event in the processor. This event varies for every architecture. Therefore,
+we use `ocperf`, part of the PMU-tools Project, to wrap all the information
+required for `perf` to generate the :file:`perf.data` file correctly for any
+Intel architecture. You can use either the `ocperf` tool or just the `perf`
+tool.
+
+.. code-block:: console
```
Comment (rcaballeromx) [inline source/clear-linux/tutorials/autofdo.rst]: +1

**Comment by iphutch** [comment]
@rcaballeromx see latest commit. I have applied these changes.

**Comment by iphutch** [comment]
Had to approve my own changes because I had requested them.

**Comment by rcaballeromx** [comment]
An updated PR has been uploaded.

**Pull request closed without merging**